src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/CSharpConvertBinder.cs

```diff
@@ -121,7 +122,13 @@ public override DynamicMetaObject FallbackConvert(DynamicMetaObject target, Dyna
             {
                 return com;
             }
+#else
+            if (!BinderHelper.IsWindowsRuntimeObject(target) && target.LimitType.IsCOMObject)
+            {
+                throw ErrorHandling.Error(ErrorCode.ERR_DynamicBindingComUnsupported);
+            }
 #endif
```

Comment:
Are we able to test these new code paths?

Reply:
I've added a test using some of the Shell COM APIs.